core/src/network/reducers/get.rs

```diff
@@ -0,0 +1,436 @@
+use crate::{
+    action::{ActionWrapper, GetEntryKey, GetLinksKey, QueryKey},
+    network::{
+        query::{GetLinksNetworkQuery, NetworkQuery},
+        reducers::send,
+        state::NetworkState,
+    },
+    state::State,
+};
+use holochain_core_types::{crud_status::CrudStatus, error::HolochainError};
+use holochain_json_api::json::JsonString;
+use holochain_persistence_api::hash::HashString;
+use lib3h_protocol::{data_types::QueryEntryData, protocol_client::Lib3hClientProtocol};
+use std::convert::TryInto;
+
+fn reduce_get_entry_inner(
+    network_state: &mut NetworkState,
+    key: &GetEntryKey,
+) -> Result<(), HolochainError> {
+    network_state.initialized()?;
+    let query_json: JsonString = NetworkQuery::GetEntry.into();
+    send(
+        network_state,
+        Lib3hClientProtocol::QueryEntry(QueryEntryData {
+            requester_agent_id: network_state.agent_id.clone().unwrap().into(),
+            request_id: key.id.clone(),
+            space_address: network_state.dna_address.clone().unwrap(),
+            entry_address: key.address.clone(),
+            query: query_json.to_string().into_bytes(),
+        }),
+    )
+}
+
+pub fn reduce_get(
+    network_state: &mut NetworkState,
+    _root_state: &State,
+    action_wrapper: &ActionWrapper,
+) {
+    let action = action_wrapper.action();
+    let (key_type, payload) = unwrap_to!(action => crate::action::Action::Query);
+    let result = match key_type {
```

Comment:
Why do we need to match on the key_type here? Can't we just pass the query we've got to over to the network as it is?